File: backend/GestionSetlistApp/Models/Membre.cs
namespace GestionSetlistApp.Models
{
    public class Membre
    {
        public int MembreId { get; set; }
        public required string Nom { get; set; }
        public required string Prenom { get; set; }
        public required DateTime DateNaissance { get; set; }
        public int Age { get; set; }
        public ICollection<MembreSetlist> MembreSetlist { get; set; } = [];
        public ICollection<MembreJoueDe> Instruments { get; set; } = [];
        public void CalculerAge()
        {
            var today = DateTime.Today;
            var age = today.Year - DateNaissance.Year;
            if (DateNaissance.Date > today.AddYears(-age))
            {
                age--;
            }
            Age = age;
        }
    }
}

File: frontend/gestion-setlist/src/app/models/membre.ts
import { MembreJoueDe } from "./membreJoueDe";
import { MembreSetlist } from "./membreSetlist";

export interface Membre {
    membreId: number;
    nom: string;
    prenom: string;
    dateNaissance: Date;
    age: number;
    membreSetlists: MembreSetlist[];
    instruments: MembreJoueDe[];
}